src/rust/engine/src/context.rs

```diff
@@ -205,7 +205,9 @@ impl Core {
             // need to take an option all the way down here and into the remote::CommandRunner struct.
             Platform::Linux,
             executor.clone(),
-            std::time::Duration::from_secs(320),
+            // The queue buffer time is added to the server-side enforced timeout to ensure that we
+            // tend to see an error from the server before the client gives up.
+            std::time::Duration::from_secs(900),
```

Comment:
Oh, this makes a lot more sense. To confirm, the server is responsible for timeouts now, whereas the client was doing it?

Reply:
Yes, but given that the client-side timeout code exists and _would_ catch strange protocol level issues, it felt reasonable to leave it.